src/Controller.js

```diff
@@ -0,0 +1,140 @@
+const inputView = require('./InputView');
+const outputView = require('./OutputView');
+const { BRIDGE } = require('../constants/game.constants');
+const InputValidator = require('../validators/InputValidator');
+const { makeBridge } = require('./BridgeMaker');
+const { generate } = require('./BridgeRandomNumberGenerator');
+const BridgeGame = require('./BridgeGame');
+
+class Controller {
+  #bridgeGame;
+
+  constructor () {
+    this.#bridgeGame = new BridgeGame();
+  }
+
+  maker () {
+    inputView.readBridgeSize(this.handleBridgeLength.bind(this));
+  }
+
+  makeAndSaveBridge (length) {
+    const bridge = makeBridge(length, generate);
+    this.#bridgeGame.setBridge(bridge);
+  }
+
+  handleBridgeLength (length) {
+    try {
+      InputValidator.isRightBridgeLength(length);
+      outputView.printBlankLine();
+      this.makeAndSaveBridge(length);
+      this.moveBrige();
+    } catch (error) {
+      outputView.printError(error);
+      this.maker();
+    }
+  }
+
+  getUpDownTotalPrintList () {
+    return `${this.getPrintList(
+      this.#bridgeGame.getUpList(),
+    )}\n${this.getPrintList(this.#bridgeGame.getDownList())}`;
+  }
+
+  outputResultPrint () {
+    outputView.printResult(
+      true,
+      this.getUpDownTotalPrintList(),
+      this.#bridgeGame.getTryCount(),
+    );
+  }
+
+  moveBrige () {
+    if (this.#bridgeGame.getMoveIndex() !== this.#bridgeGame.getBridge().length) {
+      inputView.readMoving(this.handleMoveBridge.bind(this));
+    } else {
+      this.outputResultPrint();
+    }
+  }
+
+  drawBridgeMap (string) {
+    this.#bridgeGame.move(string);
+    this.setMapArray(this.sameBridge(this.#bridgeGame.getMoveIndex()), string);
+    outputView.printMap(
+      this.getUpDownTotalPrintList(),
+    );
+  }
+
+  handleMoveInput (string) {
+    if (this.#bridgeGame.getBridge()[this.#bridgeGame.getMoveIndex()] === string) {
+      this.#bridgeGame.incrementMoveIndex();
+      return this.moveBrige();
+    }
+    this.askRetryGame(this);
+  }
+
+  handleMoveBridge (string) {
+    try {
+      InputValidator.isUpDown(string);
+      this.drawBridgeMap(string);
+      this.handleMoveInput(string);
+    } catch (error) {
+      outputView.printError(error);
+      this.moveBrige(this);
+    }
+  }
+
+  sameBridge (idx) {
+    return this.#bridgeGame.getBridge()[idx]
+    === this.#bridgeGame.getInputUpDown()
+      ? BRIDGE.ABLE
+      : BRIDGE.UNABLE;
+  }
+
+  setMapArray (sameResult, string) {
+    if (string === BRIDGE.UP && sameResult) {
+      this.#bridgeGame.getUpList().push(sameResult);
+      this.#bridgeGame.getDownList().push(' ');
+    }
+    if (string === BRIDGE.DOWN && sameResult) {
+      this.#bridgeGame.getUpList().push(' ');
+      this.#bridgeGame.getDownList().push(sameResult);
+    }
+  }
```

Comment:
메서드 이름을 `setMapArray` 보다는 `updateMapArray`가 나을 것 같습니다!

`set` 이라는 접두사는 값을 설정하는 느낌이기 때문에,
`update`가 나을 것 같습니다!

Reply:
`sameResult` 라는 이름은 어떤값이 들어있는지 유추하기가 어려운 것 같아요 😭

저는 `marker` 라는 단어를 사용했었는데, 메소드나 변수 명이 조금 길더라도 확실히 어떤 값이 있는지 나타내는게 중요하다고 합니다! 😁

P.S. 저도 네이밍하는데 아직 많이 부족해서 노력하는 중입니다 😭😭

Reply:
위 코드를 보기쉽게 좀 줄여보자면, 이렇게 짜볼 수 있을 것 같아요!

```suggestion
  setMapArray (sameResult, string) {
    this.#bridgeGame.getUpList()
        .push(string === BRIDGE.UP?sameResult:' ');
    this.#bridgeGame.getDownList()
        .push(string === BRIDGE.DOWN?sameResult:' ');
  }
```

Reply:
하나 더 첨언하자면 bridge를 update하는 로직은 bridgeGame에서 처리하는 것이 좋아보입니다! 
upList와 downList는 private 필드인데 get해서 controller에서 update하는 것보다 bridgeGame에게 시키는 것이 어떨까요?

참고) 3주차 피드백의 `객체는 객체스럽게 사용한다`
>  데이터를 꺼내지(get) 말고 메시지를 던지도록 구조를 바꿔 데이터를 가지는 객체가 일하도록 한다.

Reply:
> bridge를 update하는 로직은 bridgeGame에서 처리하는 것이 좋아보입니다!
upList와 downList는 private 필드인데 get해서 controller에서 update하는 것보다 bridgeGame에게 시키는 것이 어떨까요?

그러네요 이 부분은 bridgeGame에서 처리하는게 더 옳다고 느껴지네요!
'객체는 객체스럽게 사용한다' 덕분에 한번 더 리마인드 하고 넘어갑니다 낭경님 말씀해주신대로 리팩토링 해봐야겠어요 ☺️